File: back/ChatService.Domain/Models/ChatMedia.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChatService.Domain.Models
{
    public class ChatMedia
    {
        public int Id { get; set; }

        public int ChatId { get; set; }
        public Chat Chat { get; set; } = null!;

        public string Bucket { get; set; } = null!;

        // 🔑 уникальный ключ файла в MinIO
        public string FileKey { get; set; } = null!;

        // 📄 оригинальное имя (для UI)
        public string OriginalName { get; set; } = null!;

        // 🧩 тип (avatar, post, video и т.д.)
        public string MediaType { get; set; } = null!;

        // 🎞 mime type
        public string ContentType { get; set; } = null!;

        // 📦 размер
        public long Size { get; set; }

        public DateTime CreatedAt { get; set; }
    }
}

File: front/Сonsumer/src/utils/chatMedia.ts
import type { ChatMediaDto } from "@/interface/DTO/chat/ChatMediaDto";
import type { Media } from "@/interface/models/profile/Media";
import { MediaType } from "@/interface/models/profile/MediaType";

export function chatMediaToGallery(item: ChatMediaDto): Media {
  return {
    id: item.id,
    fileKey: "",
    bucket: "",
    mediaType:
      item.mediaType.toLowerCase() === "video" ? MediaType.VIDEO : MediaType.AVATAR,
    contentType: "",
    url: item.url,
    createdAt: item.createdAt,
  };
}

export function chatMediaListToGallery(items: ChatMediaDto[]): Media[] {
  return items.map(chatMediaToGallery);
}
